cmd/agent/subcommands/snmp/command.go

```diff
@@ -29,272 +30,267 @@ import (
 )
 
 const (
-	defaultVersion = ""
-	defaultOID     = ""
-	defaultPort    = 161
-
-	// snmp v1 & v2c
-	defaultCommunityString = "public"
-
-	// snmp v3
-	defaultUserName      = ""
-	defaultAuthProtocol  = ""
-	defaultAuthKey       = ""
-	defaultPrivProtocol  = ""
-	defaultPrivKey       = ""
-	defaultContext       = ""
-	defaultSecurityLevel = ""
-
-	// general communication options
+	defaultPort                    = 161
+	defaultCommunityString         = "public"
 	defaultTimeout                 = 10 // Timeout better suited to walking
 	defaultRetries                 = 3
 	defaultUseUnconnectedUDPSocket = false
 )
 
-// cliParams are the command-line arguments for this subcommand
-type cliParams struct {
-	*command.GlobalParams
-
-	// args are the positional command-line arguments
-	args []string
+// // connectionParams are the data needed to connect to an SNMP instance.
+type connectionParams struct {
+	// embed a SNMPConfig because it's all the same fields anyway
+	parse.SNMPConfig
+	// fields that aren't part of parse.SNMPConfig
+	SecurityLevel           string
+	UseUnconnectedUDPSocket bool
+}
 
-	// cmd is the cobra Command, used to show help
-	cmd *cobra.Command
+// argsType is an alias so we can inject the args via fx.
+type argsType []string
 
-	// general
-	snmpVersion string
+// configErr wraps any error caused by invalid configuration.
+// If the main script returns a configErr it will print the usage string along
+// with the error message.
+type configErr struct {
+	err error
+}
 
-	// v1 & v2c
-	communityString string
+func (u configErr) Error() string {
+	if u.err != nil {
+		return u.err.Error()
+	}
+	return "configuration error"
+}
 
-	// v3
-	user          string
-	authProt      string
-	authKey       string
-	privProt      string
-	privKey       string
-	snmpContext   string
-	securityLevel string
+func (u configErr) Unwrap() error {
+	return u.err
+}
 
-	// communication
-	retries              int
-	timeout              int
-	unconnectedUDPSocket bool
+// confErrf is a shorthand for creating a simple confErr.
+func confErrf(msg string, args ...any) configErr {
+	return configErr{fmt.Errorf(msg, args...)}
 }
 
 // Commands returns a slice of subcommands for the 'agent' command.
 func Commands(globalParams *command.GlobalParams) []*cobra.Command {
-	cliParams := &cliParams{
-		GlobalParams: globalParams,
+	connParams := &connectionParams{}
+	snmpCmd := &cobra.Command{
+		Use:   "snmp",
+		Short: "Snmp tools",
+		Long:  ``,
 	}
+
 	snmpWalkCmd := &cobra.Command{
-		Use:   "walk <IP Address>[:Port] [OID] [OPTIONS]",
-		Short: "Perform a snmpwalk, if only a valid IP address is provided with the oid then the agent default snmp config will be used",
+		Use:   "walk <IP Address>[:Port] [OID]",
```

Comment:
why this change since the walk takes other options in addition to the ip address port and OID ? (the original `snmpwalk` command also refers to the rest of the flag as options)

Reply:
Cobra adds `[flags]` to the end of the help string automatically, so it was showing as
```
Usage:
  agent snmp walk <IP Address>[:Port] [OID] [OPTIONS] [flags]
  ```